scripts/fota_builder.py

```diff
@@ -0,0 +1,363 @@
+#!/usr/bin/env python3
+""" Create fota archive """
+import json
+import os
+import shutil
+import subprocess
+import argparse
+import stat
+import sys
+from pathlib import Path
+from datetime import datetime
+import yaml
+
+EINVAL = 22
+
+
+class FotaError(Exception):
+    """Exception raised for errors during creating fota archicve
+
+    Attributes:
+        error code -- Error code
+        message -- explanation of the error
+    """
+
+    def __init__(self, error_code, message="Fota error"):
+        self.code = error_code
+        self.message = message
+        super().__init__(self.message)
+
+
+class FotaBuilder:
+    """Create fota archive class"""
+
+    def __init__(self, conf, verbose) -> None:
+        self.conf = conf
```

Comment:
Not all memebers and methods are public.  Make corresponding members and methods as private:

https://www.geeksforgeeks.org/private-methods-in-python/
https://www.geeksforgeeks.org/private-variables-python/